### PR TITLE
Additional IPU test coverage

### PR DESCRIPTION
**Pull request by s-maddrellmander**
## Changelogs

Good to land asap - IPU tests with pop torch being run properly now. 

- Adding more tests to cover IPU code that's not currently covered
- Regression: Skipping the IPU data loader tests - need to get IPU model working for these tests - will open a new PR for this next
- Fixing the Github action testing env

---

_Checklist:_

- [x] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [x] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [x] _Update the API documentation is a new function is added, or an existing one is deleted._
- [x] _Write concise and explanatory changelogs above._
- [x] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

### CODE REVIEW

**Comment by DomInvivo** [comment]
@s-maddrellmander , is there a test that runs a full model on the ipu? Maybe it's already there?

**Comment by s-maddrellmander** [comment]
Coverage decrease expected - need the IPU tests running the whole model handled differently

File: tests/config_test_ipu_dataloader_multitask.yaml [hunk 1]
```diff
@@ -0,0 +1,346 @@
+# Testing the gcn model with the PCQMv2 dataset on IPU.
+constants:
+  name: &name neurips2023_small_data_gcn
```
Comment (zhiyil-graphcore) [inline tests/config_test_ipu_dataloader_multitask.yaml:3]: This config is just data-loading right? I assume there is not a way to run the full model on IPU without physical IPU?

**Review by zhiyil-graphcore — COMMENTED** [review]
Nice!

File: tests/config_test_ipu_dataloader_multitask.yaml (hunk 1 above)
Comment (s-maddrellmander) [reply tests/config_test_ipu_dataloader_multitask.yaml:3]: Those tests were also compiling the model and starting training, which while okay on a machine with IPUs, not so good on the runner. So I'm skipping the "data loader" tests for now, while we get the IPU model working with lightning. (In a different PR)

**Comment by s-maddrellmander** [comment]
@hadim - would it be possible to get a review for this please? Would like to get the IPU tests passing on the main branch so we can keep updating + run the new tests as they are being developed.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -50,7 +50,7 @@ jobs:
         run: python -m pip install --no-deps -e . # `-e` required for correct `coverage` run.
 
       - name: Run tests
-        run: pytest
+        run: pytest -m 'not ipu'
```
Comment (hadim) [inline .github/workflows/test.yml:53]: Do you think that flag should be added in `pyproject.toml` so people running the tests locally (without IPU) will work (and also match the CI behavior)?

**Review by hadim — COMMENTED** [review]
LGTM.

See my question below.

**Comment by hadim** [comment]
And congrats on 

image


:-D

File: .github/workflows/test.yml (hunk 2 above)
Comment (s-maddrellmander) [reply .github/workflows/test.yml:53]: This is a good idea - I'll check how it works with overwriting

**Pull request merged**